cub/device/dispatch/dispatch_radix_sort.cuh

```diff
@@ -1281,9 +1305,9 @@ struct DispatchRadixSort :
             // lookback
             max_num_blocks * RADIX_DIGITS * sizeof(AtomicOffsetT),
             // extra key buffer
-            is_overwrite_okay || num_passes <= 1 ? 0 : num_items * sizeof(KeyT),
+            num_passes <= 1 ? 0 : num_items * sizeof(KeyT),
```

Comment:
This will drastically increase the amount of temporary storage needed for some invocations of radix sort. We need to keep this optimization in place for the `DoubleBuffer` overloads in the `Device*RadixSort` APIs, since folks specifically use those to reduce the temporary storage allocations.